docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,105 @@
+# Syncing LDAP Groups to OpenShift Groups
+
+Linking OpenShift `Groups` to their representations on LDAP servers allows for administrators to have an easier way to manage `Groups` inside OpenShift. This additional functionality will require no additional fields on the OpenShift `Group` object. The sync job will indicate that an OpenShift `Group` mirrors an external representation by setting the `openshift.io/externalRecordProvider` `Annotation` in the `Group` as well as adding a `Label` such as `openshift.io/externalRecordProvider/LDAP` to the `Group`. Other information (e.g. last sync time or resource version) will be stored as `Annotations` on the OpenShift `Group`. Logistical information for the LDAP sync will be stored in an `LDAPSyncOptions` that the administrator will need to provide.
```

Comment:
```
Cluster admins need a way to synchronize their LDAP group membership to OpenShift groups.  This requires being able to describe which LDAP groups need to be sync'ed, mapping LDAP users to Openshift users, mapping LDAP group names to openshift groups names, and removing orphaned openshift groups.
```

No reason to into details here.  Give a brief overview of what needs doing and what capability is required.